tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,779 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const Bluebird = require('bluebird');
+const request = Bluebird.promisifyAll(require('request'));
+const nock = require('nock');
+const s3Packages = require('../../lib/shared/s3-packages');
+
+describe('Shared: s3Packages', function() {
+
+  describe('.getBucketUrlFromVersion()', function() {
+
+    describe('given the version has a short git commit hash build number', function() {
+
+      it('should return the snapshot s3 bucket', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0+6374412');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_SNAPSHOT);
+      });
+
+      it('should return the snapshot s3 bucket if the version has a pre release tag', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0-beta.19+6374412');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_SNAPSHOT);
+      });
+
+    });
+
+    describe('given the version has a short git commit hash build number', function() {
+
+      it('should return the snapshot s3 bucket', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0+6374412554b034799bfc6e13b4e39c3f5e6386e6');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_SNAPSHOT);
+      });
+
+      it('should return the snapshot s3 bucket if the version has a pre release', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0-beta.19+6374412554b034799bfc6e13b4e39c3f5e6386e6');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_SNAPSHOT);
+      });
+
+    });
+
+    describe('given the version has no build number', function() {
+
+      it('should return the production s3 bucket', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_PRODUCTION);
+      });
+
+      it('should return the production s3 bucket if the version has a pre release tag', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0-beta.19');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_PRODUCTION);
+      });
+
+    });
+
+    describe('given a build number that is not a git commit hash', function() {
```

Comment:
I guess if we went with my `.getReleaseTypeFromVersion` we could maybe return 'unknown' (or 'custom') in this case? (and then skip the update-check, given that 'unknown' won't exist in `s3Packages.BUCKET_URLS`).

Perfect for custom-built debug builds, that we don't want test users to automatically get auto-updated from ;-)